pynocular/backends/memory.py

```diff
@@ -0,0 +1,239 @@
+        """Create a new transaction
+
+        This fails as a warning that the in-memory backend does not support transactions.
+        """
+        raise NotImplementedError()
+
+    async def select(
+        self,
+        config: DatabaseModelConfig,
+        where_expressions: Optional[List[BinaryExpression]] = None,
+        order_by: Optional[List[UnaryExpression]] = None,
+        limit: Optional[int] = None,
+    ) -> List[Dict[str, Any]]:
+        """Select a group of records
+
+        Args:
+            config: DatabaseModelConfig instance that contains references to a table and
+                columns that can be used to build queries suited to the backend.
+            where_expressions: A list of BinaryExpressions for the table that will be
+                `and`ed together for the where clause of the backend query
+            order_by: A list of criteria for the order_by clause
+            limit: The number of records to return
+
+        Returns:
+            list of records
+
+        Raises:
+            InvalidFieldValue: The class is missing a database table
+
+        """
+        records = self.records[config.table.name]
+
+        if where_expressions:
+            records = [
+                record
+                for record in records
+                if all(
+                    evaluate_column_element(expr, record) for expr in where_expressions
+                )
+            ]
+
+        if order_by:
+            for expr in order_by:
+                if isinstance(expr, UnaryExpression):
+                    column = expr.element
+                    reverse = expr.modifier == desc_op
+                else:
+                    # Assume a column was provided with no explicit sorting modifier
+                    column = expr
+                    reverse = False
+
+                records = sorted(
+                    records, key=lambda r: r.get(column.name), reverse=reverse
+                )
+
+        if limit is None:
+            records[:limit]
+
+        return records
+
+    async def create_records(
+        self, config: DatabaseModelConfig, records: List[Dict[str, Any]]
+    ) -> List[Dict[str, Any]]:
+        """Create new group of records
+
+        Args:
+            config: DatabaseModelConfig instance that contains references to a table and
+                columns that can be used to build queries suited to the backend.
+            records: List of records to persist
+
+        Returns:
+            list of newly created records
+
+        """
+        for record in records:
+            for primary_key in config.primary_keys:
+                value = (
+                    next(self._pk_generator)
+                    if isinstance(primary_key.type, Integer)
+                    else str(uuid4())
+                )
+                record.setdefault(primary_key.name, value)
+
+        self.records[config.table.name].extend(records)
+
+        return self.records[config.table.name]
```

Comment:
one thing I ran into with V1 is with the patch approach (now in memory) it doesn't handle things that are flagged as `fetch_on_create`. Thats fine for now but we may want to think about supporting that in the future

Reply:
I added support for fields that need fetch_on_create  or fetch_on_update when the type is datetime

Reply:
oh cool! I dont see it here when a record is created or when they are updated though. I do see it in `upsert`

Reply:
Ahh ok i missed that. I can do a follow up